### ISSUE TITLE
Implement auth for apiserver's use of HostNetwork

### ISSUE THREAD

**Issue by vmarmol** [GoogleCloudPlatform/kubernetes#5944]
Today the apiserver does not verify that requests for HostNetwork usage can only be allowed if the capability is set. The Kubelet checks once the pod is bound however.

We need to add the check in validation and plumb the flag through the apiserver.

**Comment by davidopp** [comment]
@vmarmol should this be considered a 1.0 blocker?

**Comment by vmarmol** [comment]
I don't think so. The behavior will be a bit weird without it: the apiserver schedules the pod, the Kubelet rejects it. But it should be rare enough that we can live with it for now. The users that want it will be setting up their customized clusters anyways.

**Comment by smarterclayton** [comment]
@pweil- we need to fold this into SCC as discussed

**Comment by pweil-** [comment]
@smarterclayton - agree, it's on my list of follow ups for SCCs

**Comment by erictune** [comment]
This is in PodSecurityPolicy (the new name for SCC).  Closing this in favor of https://github.com/kubernetes/kubernetes/issues/23217
